File: Model/Moto.cs
using MotoScript_Challange.Model;
using System.ComponentModel.DataAnnotations;
using System.Runtime.InteropServices;

namespace MotoScript_Challange.Model
{
    public class Moto
    {
        [Key]
        public long id { get; set; }
        [Required]
        [MaxLength(100)]
        public string nome { get; set; }
        [Required]
        [MaxLength(200)]
        public string descricao { get; set; }
        [Required]
        public bool prontaParaUso { get; set; }
        [Required]
        public bool semPlaca { get; set; }
        [Required]
        public string chassi { get; set; }
        [Required]
        public string motor { get; set; }

        public Moto()
        {
            id = 0;
            nome = string.Empty;
            descricao = string.Empty;
            prontaParaUso = false;
            semPlaca = false;
            chassi = string.Empty;
            motor = string.Empty;
        }
    }
}

File: PROJETO/src/main/java/com/example/projeto/model/Moto.java
package com.example.projeto.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Moto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "O nome é obrigatório")
    private String nome;

    @NotBlank(message = "A descrição é obrigatória")
    @Pattern(regexp = "^[A-Z].*", message = "Deve começar com letra maiúscula")
    private String descricao;

    private boolean prontaParaUso;
    private boolean semPlaca;
    private String chassi;
    private String motor;

}
